src/Tile_test.py

```diff
@@ -0,0 +1,26 @@
+class TestTile:
+    def test_building(self):
+        from Tile import Tile
+        from Terrain import Terrain
+        from Building import Building
+
+        spot = Tile(Terrain.MOUNTAIN)
+        assert spot.can_build(Building.DWELLING)
+        assert not spot.can_build(Building.TRADING_HOUSE)
+        spot.build(Building.DWELLING)
+        assert spot.can_build(Building.TRADING_HOUSE)
+        assert not spot.can_build(Building.DWELLING)
+        assert not spot.can_build(Building.SANCTUARY)
+        spot.build(Building.TRADING_HOUSE)
+        assert spot.can_build(Building.TEMPLE)
+        assert spot.can_build(Building.STRONGHOLD)
+        assert not spot.can_build(Building.DWELLING)
```

Comment:
Can I suggest adding `not spot.can_build(Building.TRADING_HOUSE)` here too and the similar cases below? (I see that `Building.DWELLING` has it above)